src/jit/codegenxarch.cpp

```diff
@@ -4641,6 +4641,15 @@ void CodeGen::genCallInstruction(GenTreePtr node)
     // all virtuals should have been expanded into a control expression
     assert(!call->IsVirtual() || call->gtControlExpr || call->gtCallAddr);
 
+    // Insert a GS check if necessary
```

Comment:
You might want to add a brief comment that this must be done before consuming the registers, as their liveness must extend to the actual call.